src/gui/my_controls.cpp

```diff
@@ -1527,63 +1527,54 @@ ClassVolumeSelectPanel::ClassVolumeSelectPanel(wxWindow* parent, wxWindowID id,
 ClassVolumeSelectPanel::~ClassVolumeSelectPanel( ) {
 }
 
+CombinedPackageClassSelectionPanel::CombinedPackageClassSelectionPanel(wxWindow* parent, wxWindowID id, const wxPoint& pos, const wxSize& size, long style)
+    : wxPanel(parent, id, pos, size, style) {
+    MainSizer = new wxBoxSizer(wxHORIZONTAL);
```

Comment:
Should there be a corresponding delete in the destructor for this allocation by new? I see several instances of MainSizer = new wxBoxSizer in several files, none with deletes. I'm not sure if I'm missing some polymorphic weirdness, or if it is just a small allocation that is never freed until the GUI closes and we've never noticed. If the latter, it could be a separate issue to add the deletes.

Reply:
So there are a few weird aspects of this; the first thing that's important to note is I prefer to keep the top-level objects on the stack where possible (ResampleDialog being the "top-level" in this instance) so that the highest level of memory for an object is managed simply by the object's scope. The reason this matters is because of the other relevant factor, which is that when the top-level object is deleted or going out of scope on the stack, any sizers which are pointers will automatically deallocate pointer-based children, and the top-level object will delete the sizers themselves (I think all via smart pointers, as I've read the maintainers of wxWidgets use these for base classes).

So, there is some implicit trusting going on in assuming that wxWidgets has properly implemented smart pointers so this works how I think it does. Therefore it could be worth opening an issue at some point to add the extra safety of deletions, or at the very least verifying that wxWidgets 3.0.5 utilizes smart pointers for the base classes.

Reply:
okay, thanks for the info. It looks like operator new in these cases is an overload written in wx/memory.h, and I see [here ](https://wiki.wxwidgets.org/Avoiding_Memory_Leaks) that 

> Classes that are not derived from wxWindow or wxSizer

> Classes that are not derived off wxWindow or wxSizer can be created and deleted like normal C++ classes, as described above, either on the stack or on the heap. 

So it sounds like for wxWindow/wxSizer, and for their derived classes, WX is handling memory cleanup. I don't quite see how they are making that work, but this is good enough for me I guess.